### PR TITLE
Fix initial report to SCM call when using link_package step

### LINKED ISSUE #12121: SCM/CI: Report pending status for multibuilds
We are about to use the SCM/CI for openQA projects.
We are using multibuilds, and it is a documented limitation, that multibuilds don't report a pending status, but only report the final fail or success.
https://openbuildservice.org/help/manuals/obs-user-guide/cha.obs.scm_ci_workflow_integration.html#sec.obs.obs_scm_ci_workflow_integration.setup.status_reporting

This can be problematic because when looking at a PR, one doesn't even know if there are still running checks, and we have a lot of checks.
A PR might get merged because every check passes.
We also use mergify to automatically merge in certain conditions, and we have to manually add every OBS build/platform to the configuration to be sure that it doesn't merge before those checks are reported.

So it would be nice if the pending status could also be reported.

### PR DESCRIPTION
**Pull request by vpereira**
While trying to fix #12121 we found 2 issues:

- `Package#file_exists?` doesn't expand the files on backend and therefore it cannot find the "_multibuild" file

-  ~~When we try to call `report_to_scm` on the `LinkPackageStep` the repositories aren't configured and therefore the call to `workflow_repositories` done inside of `report_to_scm` fails.. We have to call `report_to_scm` inside `ConfigureRepositoriesStep`~~

EDIT: In this PR me and @dmarcoux decided to fix just one issue. The `report_to_scm` problem will be handled differently

### CODE REVIEW

File: src/api/app/models/package.rb [hunk 1]
```diff
@@ -1331,7 +1331,7 @@ def backend_build_command(command, build_project, params)
   end
 
   def file_exists?(filename)
-    dir_hash.key?('entry') && [dir_hash['entry']].flatten.any? { |item| item['name'] == filename }
+    dir_hash.key?('entry') && [dir_hash(expand: 1)['entry']].flatten.any? { |item| item['name'] == filename }
```
Comment (vpereira) [inline src/api/app/models/package.rb:1334]:
```
# How it was called originally
[9] pry(main)> pkg.dir_hash
[backend] GET: /source/home:Admin:link:vpereira:hello_world:PR-62/hello_world
request took 0.006381454 0.020073037000000002
=> {"name"=>"hello_world",
 "rev"=>"2",
 "vrev"=>"2",
 "srcmd5"=>"2d095ef94eca1960141228046dbe598c",
 "linkinfo"=>
  {"project"=>"home:Admin",
   "package"=>"hello_world",
   "srcmd5"=>"6cf97d151ff845f7aaec5f9ed3f0c105",
   "xsrcmd5"=>"b61bfd162ad91c85375867f1caf1f4e5",
   "lsrcmd5"=>"2d095ef94eca1960141228046dbe598c"},
 "serviceinfo"=>{"code"=>"succeeded", "xsrcmd5"=>"e29c775a7fadedf2482176ff24113eb4"},
 "entry"=>
  [{"name"=>"_branch_request", "md5"=>"1b092426c216175c1a78eeb10cb8c46e", "size"=>"138", "mtime"=>"1643374049"},
   {"name"=>"_link", "md5"=>"1189bcb57f87db8ec51ab130f393eb3f", "size"=>"50", "mtime"=>"1643367249"}]}
# How it is called now
[10] pry(main)> pkg.dir_hash(expand: 1)
[backend] GET: /source/home:Admin:link:vpereira:hello_world:PR-62/hello_world?expand=1
request took 0.022222506 0.042295543000000005
=> {"name"=>"hello_world",
 "rev"=>"7c00b64894ed0cfe3f76136e3c08e295",
 "vrev"=>"3",
 "srcmd5"=>"7c00b64894ed0cfe3f76136e3c08e295",
 "linkinfo"=>
  {"project"=>"home:Admin",
   "package"=>"hello_world",
   "srcmd5"=>"6cf97d151ff845f7aaec5f9ed3f0c105",
   "lsrcmd5"=>"e29c775a7fadedf2482176ff24113eb4"},
 "serviceinfo"=>{"code"=>"succeeded", "lsrcmd5"=>"2d095ef94eca1960141228046dbe598c"},
 "entry"=>
  [{"name"=>"_branch_request", "md5"=>"1b092426c216175c1a78eeb10cb8c46e", "size"=>"138", "mtime"=>"1643374049"},
   {"name"=>"_multibuild", "md5"=>"860006b29f5b86f820d6676571c71e7e", "size"=>"63", "mtime"=>"1643366931"},
   {"name"=>"_service", "md5"=>"32d343afe2c84734694e21afb8ef2a8e", "size"=>"331", "mtime"=>"1643366931"},
   {"name"=>"_service:obs_scm:hello_world-1643374042.1ece639.obscpio",
    "md5"=>"7957dde05d7ef261b101a2e8a0ff2ac0",
    "size"=>"2057",
    "mtime"=>"1643374051"},
   {"name"=>"_service:obs_scm:hello_world.obsinfo", "md5"=>"187a7b6271965e3579b5265092fe5362", "size"=>"113", "mtime"=>"1643374051"},
   {"name"=>"_service:obs_scm:hello_world.spec", "md5"=>"595f687689f26fb2c6e9be724d94e3fd", "size"=>"459", "mtime"=>"1643367250"}]}
[11] pry(main)> reload!
```

File: src/api/app/models/workflow/step/configure_repositories.rb [hunk 2]
```diff
@@ -25,6 +25,7 @@ def call(_options = {})
       end
     end
 
+    report_to_scm({})
```
Comment (vpereira) [inline src/api/app/models/workflow/step/configure_repositories.rb:28]: We don't have access to the filters here, but is it necessary?

When it is called it breaks with:

```
33a915bc-d8d5-4fe3-9f80-ae427817a5d3] [52:2015.68]   
[33a915bc-d8d5-4fe3-9f80-ae427817a5d3] NoMethodError (undefined method `start_with?' for nil:NilClass):
[33a915bc-d8d5-4fe3-9f80-ae427817a5d3]   
[33a915bc-d8d5-4fe3-9f80-ae427817a5d3] app/models/concerns/multibuild_package.rb:11:in `striping_multibuild_suffix'
[33a915bc-d8d5-4fe3-9f80-ae427817a5d3] app/models/package.rb:157:in `get_by_project_and_name'
[33a915bc-d8d5-4fe3-9f80-ae427817a5d3] app/models/workflow/step.rb:34:in `target_package'
[33a915bc-d8d5-4fe3-9f80-ae427817a5d3] app/models/workflow/step.rb:99:in `multibuild_flavors'
[33a915bc-d8d5-4fe3-9f80-ae427817a5d3] app/models/workflow/step.rb:75:in `target_package_names'
[33a915bc-d8d5-4fe3-9f80-ae427817a5d3] app/models/workflow/step.rb:171:in `block (2 levels) in report_to_scm'
[33a915bc-d8d5-4fe3-9f80-ae427817a5d3] app/models/workflow/step.rb:170:in `block in report_to_scm'
[33a915bc-d8d5-4fe3-9f80-ae427817a5d3] app/models/workflow/step.rb:168:in `report_to_scm'
[33a915bc-d8d5-4fe3-9f80-ae427817a5d3] app/models/workflow/step/configure_repositories.rb:29:in `call'
[33a915bc-d8d5-4fe3-9f80-ae427817a5d3] app/models/workflow.rb:34:in `block in call'
[33a915bc-d8d5-4fe3-9f80-ae427817a5d3] app/models/workflow.rb:33:in `each'
[33a915bc-d8d5-4fe3-9f80-ae427817a5d3] app/models/workflow.rb:33:in `call'
[33a915bc-d8d5-4fe3-9f80-ae427817a5d3] app/models/token/workflow.rb:20:in `each'
[33a915bc-d8d5-4fe3-9f80-ae427817a5d3] app/models/token/workflow.rb:20:in `call'
[33a915bc-d8d5-4fe3-9f80-ae427817a5d3] app/controllers/trigger_workflow_controller.rb:12:in `block in create'
[33a915bc-d8d5-4fe3-9f80-ae427817a5d3] app/models/user.rb:864:in `run_as'
[33a915bc-d8d5-4fe3-9f80-ae427817a5d3] app/controllers/trigger_workflow_controller.rb:11:in `create'
[33a915bc-d8d5-4fe3-9f80-ae427817a5d3] config/initializers/wrap_parameters.rb:37:in `call'
```

File: src/api/app/models/workflow/step/link_package_step.rb [hunk 3]
```diff
@@ -20,7 +20,7 @@ def link_package(workflow_filters = {})
     # SCMs don't support statuses for tags, so we don't need to report back in this case
     unless scm_webhook.tag_push_event?
       create_or_update_subscriptions(target_package, workflow_filters)
-      report_to_scm(workflow_filters)
+      # report_to_scm(workflow_filters)
```
Comment (vpereira) [inline src/api/app/models/workflow/step/link_package_step.rb:23]: There is any use-case where we still have to call `report_to_scm` here?

File: src/api/spec/models/package_spec.rb [hunk 4]
```diff
@@ -145,6 +145,12 @@
   end
 
   describe '#file_exists?' do
+    context '_multibuild file should exist' do
+      let!(:multibuild_package) { create(:multibuild_package, name: 'test', project: home_project) }
+
+      it { expect(multibuild_package.file_exists?('_multibuild')).to eq(true) }
```
Comment (vpereira) [inline src/api/spec/models/package_spec.rb:151]: test case to cover the file expansion on `file_exists?`

**Comment by vpereira** [comment]
[Fix initial reporting of multibuild flavors (M)(Day 3)](https://trello.com/c/L7saTIFK/1827-fix-initial-reporting-of-multibuild-flavors-mday-3)

**Review by hennevogel — APPROVED** [review]

**Pull request merged**